qa/vagrant/src/main/java/org/elasticsearch/packaging/util/ServerUtils.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.packaging.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.entity.ContentType;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+
+public class ServerUtils {
+
+    private static final Log LOG = LogFactory.getLog(ServerUtils.class);
+
+    private static final long waitTime = TimeUnit.SECONDS.toMillis(60);
+
+    public static void waitForElasticsearch() throws IOException {
+        waitForElasticsearch("green", null);
+    }
+
+    public static void waitForElasticsearch(String status, String index) throws IOException {
+
+        Objects.requireNonNull(status);
+
+        // we loop here rather than letting httpclient handle retries so we can measure the entire waiting time
+        final long startTime = System.currentTimeMillis();
+        long timeElapsed = 0;
+        boolean started = false;
+        while (started == false && timeElapsed < waitTime) {
+            try {
+
+                final HttpResponse response = Request.Get("http://localhost:9200/_cluster/health")
+                    .connectTimeout((int) waitTime)
```

Comment:
I would pass in a fraction of waitTime here or some smaller fixed value. 
The initial connect could hang in that timeout and there will be no retries because the outer loop timed out as well.